sdk/ai/azure-ai-generative/azure/ai/generative/index/_build_mlindex.py

```diff
@@ -154,6 +156,47 @@ def build_index(
             "azureml.mlIndexAssetSource": "AzureML Data",
             "azureml.mlIndexAssetPipelineRunId": "Local",
         }
+    if vector_store.lower() == "pinecone":
+        pinecone_args = {
+            "index_name": index_config.pinecone_index_name,
+        }
+        if not index_config.pinecone_connection_id:
+            import os
+            pinecone_args = {
+                **pinecone_args,
+                **{
+                    "environment": os.getenv("PINECONE_ENVIRONMENT")
+                }
+            }
+            connection_args = {
+                "connection_type": "environment",
+                "connection": {"key": "PINECONE_API_KEY"}
+            }
+        else:
+            pinecone_connection = get_connection_by_id_v2(index_config.pinecone_connection_id)
+            pinecone_args = {
+                **pinecone_args,
+                **{
+                    "environment": pinecone_connection["properties"]["metadata"]["environment"]
+                }
```

Comment:
import get_metadata_from_connection from azure.ai.generative.index._utils.connections

```suggestion
                    "environment": get_metadata_from_connection(pinecone_connection)["environment"]
                }
```

Reply:
updated